File: src/com/eagle/programmar/COBOL/COBOL_ReportLine.java
// Copyright Eagle Legacy Modernization LLC, 2010-date
// Original author: Steven A. O'Hara, Aug 13, 2022

package com.eagle.programmar.COBOL;

import com.eagle.programmar.COBOL.Symbols.COBOL_Identifier_Reference;
import com.eagle.programmar.COBOL.Symbols.COBOL_ReportLine_Definition;
import com.eagle.programmar.COBOL.Terminals.COBOL_Keyword;
import com.eagle.programmar.COBOL.Terminals.COBOL_KeywordChoice;
import com.eagle.programmar.COBOL.Terminals.COBOL_Level;
import com.eagle.programmar.COBOL.Terminals.COBOL_Literal;
import com.eagle.programmar.COBOL.Terminals.COBOL_Number;
import com.eagle.programmar.COBOL.Terminals.COBOL_Picture;
import com.eagle.tokens.TokenChooser;
import com.eagle.tokens.TokenList;
import com.eagle.tokens.TokenSequence;
import com.eagle.tokens.punctuation.PunctuationPeriod;

public class COBOL_ReportLine extends TokenSequence
{
	public @S(10) COBOL_Level O2;
	public @S(20) COBOL_Keyword LINE = new COBOL_Keyword("LINE");
	public @S(30) @OPT COBOL_Keyword IS = new COBOL_Keyword("IS");
	public @S(40) @OPT COBOL_Keyword PLUS = new COBOL_Keyword("PLUS");
	public @S(50) @OPT COBOL_Number line;
	public @S(60) PunctuationPeriod dot;

	public @S(70) TokenList<COBOL_ColumnLine> columnLines;

	public static class COBOL_ColumnLine extends TokenSequence
	{
		public @S(10) COBOL_Level O3;
		public @S(20) @OPT COBOL_ReportLine_Definition id;
		public @S(30) COBOL_Keyword COLUMN = new COBOL_Keyword("COLUMN");
		public @S(40) @OPT COBOL_Number column;

		public @S(50) COBOL_KeywordChoice pic = new COBOL_KeywordChoice("PIC", "PICTURE");
		public @S(60) COBOL_Picture picture;
		public @S(70) TokenList<COBOL_ColumnLineClause> columnClauses;
		public @S(80) PunctuationPeriod dot;

		public static class COBOL_ColumnLineClause extends TokenChooser
		{
			public @CHOICE static class COBOL_ColumnLineSource extends TokenSequence
			{
				public @S(10) COBOL_Keyword SOURCE = new COBOL_Keyword("SOURCE");
				public @S(20) COBOL_Identifier_Reference id;
				public @S(30) @OPT COBOL_Subscript subscript;
			}

			public @CHOICE static class COBOL_SourceClause extends TokenSequence
			{
				public @S(10) COBOL_Keyword SOURCE = new COBOL_Keyword("SOURCE");
				public @S(20) COBOL_Keyword PAGECOUNTER = new COBOL_Keyword("PAGE-COUNTER");
			}

			public @CHOICE static class COBOL_ColumnLineGroup extends TokenSequence
			{
				public @S(10) COBOL_Keyword GROUP = new COBOL_Keyword("GROUP");
				public @S(20) COBOL_Keyword INDICATE = new COBOL_Keyword("INDICATE");
			}

			public @CHOICE static class COBOL_ColumnLineValue extends TokenSequence
			{
				public @S(10) COBOL_Keyword VALUE = new COBOL_Keyword("VALUE");
				public @S(20) @OPT COBOL_Keyword ALL = new COBOL_Keyword("ALL");
				public @S(30) COBOL_Literal val;
			}

			public @CHOICE static class COBOL_ColumnLineSum extends TokenSequence
			{
				public @S(10) COBOL_Keyword SUM = new COBOL_Keyword("SUM");
				public @S(20) COBOL_Identifier_Reference id;
			}
		}
	}
}

File: csrc/com/eagle/programmar/COBOL/COBOL_ReportLine.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// Purchase a Premium Edition license at:
// https://www.tangiblesoftwaresolutions.com/order/order-java-to-csharp.html
// ====================================================================================================

// Copyright Eagle Legacy Modernization LLC, 2010-date
// Original author: Steven A. O'Hara, Aug 13, 2022

namespace com.eagle.programmar.COBOL
{
	using COBOL_Identifier_Reference = com.eagle.programmar.COBOL.Symbols.COBOL_Identifier_Reference;
	using COBOL_ReportLine_Definition = com.eagle.programmar.COBOL.Symbols.COBOL_ReportLine_Definition;
	using COBOL_Keyword = com.eagle.programmar.COBOL.Terminals.COBOL_Keyword;
	using COBOL_KeywordChoice = com.eagle.programmar.COBOL.Terminals.COBOL_KeywordChoice;
	using COBOL_Level = com.eagle.programmar.COBOL.Terminals.COBOL_Level;
	using COBOL_Literal = com.eagle.programmar.COBOL.Terminals.COBOL_Literal;
	using COBOL_Number = com.eagle.programmar.COBOL.Terminals.COBOL_Number;
	using COBOL_Picture = com.eagle.programmar.COBOL.Terminals.COBOL_Picture;
	using TokenChooser = com.eagle.tokens.TokenChooser;
	using TokenList = com.eagle.tokens.TokenList;
	using TokenSequence = com.eagle.tokens.TokenSequence;
	using PunctuationPeriod = com.eagle.tokens.punctuation.PunctuationPeriod;

	public class COBOL_ReportLine : TokenSequence
	{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.COBOL.Terminals.COBOL_Level O2;
		public COBOL_Level O2;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword LINE = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("LINE");
		public COBOL_Keyword LINE = new COBOL_Keyword("LINE");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) @OPT COBOL_Keyword IS = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("IS");
		public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(40) @OPT COBOL_Keyword PLUS = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("PLUS");
		public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(50) @OPT COBOL_Number line;
		public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(60) com.eagle.tokens.punctuation.PunctuationPeriod dot;
		public PunctuationPeriod dot;

// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(70) com.eagle.tokens.TokenList<COBOL_ColumnLine> columnLines;
		public TokenList<COBOL_ColumnLine> columnLines;

		public class COBOL_ColumnLine : TokenSequence
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.COBOL.Terminals.COBOL_Level O3;
			public COBOL_Level O3;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) @OPT COBOL_ReportLine_Definition id;
			public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword COLUMN = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("COLUMN");
			public COBOL_Keyword COLUMN = new COBOL_Keyword("COLUMN");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(40) @OPT COBOL_Number column;
			public  OPT;

// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(50) com.eagle.programmar.COBOL.Terminals.COBOL_KeywordChoice pic = new com.eagle.programmar.COBOL.Terminals.COBOL_KeywordChoice("PIC", "PICTURE");
			public COBOL_KeywordChoice pic = new COBOL_KeywordChoice("PIC", "PICTURE");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(60) com.eagle.programmar.COBOL.Terminals.COBOL_Picture picture;
			public COBOL_Picture picture;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(70) com.eagle.tokens.TokenList<COBOL_ColumnLineClause> columnClauses;
			public TokenList<COBOL_ColumnLineClause> columnClauses;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(80) com.eagle.tokens.punctuation.PunctuationPeriod dot;
			public PunctuationPeriod dot;

			public class COBOL_ColumnLineClause : TokenChooser
			{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE static class COBOL_ColumnLineSource extends com.eagle.tokens.TokenSequence
				public class COBOL_ColumnLineSource : TokenSequence
				{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword SOURCE = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("SOURCE");
					public COBOL_Keyword SOURCE = new COBOL_Keyword("SOURCE");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.COBOL.Symbols.COBOL_Identifier_Reference id;
					public COBOL_Identifier_Reference id;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) @OPT COBOL_Subscript subscript;
					public  OPT;
				}

// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE static class COBOL_SourceClause extends com.eagle.tokens.TokenSequence
				public class COBOL_SourceClause : TokenSequence
				{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword SOURCE = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("SOURCE");
					public COBOL_Keyword SOURCE = new COBOL_Keyword("SOURCE");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword PAGECOUNTER = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("PAGE-COUNTER");
					public COBOL_Keyword PAGECOUNTER = new COBOL_Keyword("PAGE-COUNTER");
				}

// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE static class COBOL_ColumnLineGroup extends com.eagle.tokens.TokenSequence
				public class COBOL_ColumnLineGroup : TokenSequence
				{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword GROUP = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("GROUP");
					public COBOL_Keyword GROUP = new COBOL_Keyword("GROUP");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword INDICATE = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("INDICATE");
					public COBOL_Keyword INDICATE = new COBOL_Keyword("INDICATE");
				}

// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE static class COBOL_ColumnLineValue extends com.eagle.tokens.TokenSequence
				public class COBOL_ColumnLineValue : TokenSequence
				{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword VALUE = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("VALUE");
					public COBOL_Keyword VALUE = new COBOL_Keyword("VALUE");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) @OPT COBOL_Keyword ALL = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("ALL");
					public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) com.eagle.programmar.COBOL.Terminals.COBOL_Literal val;
					public COBOL_Literal val;
				}

// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE static class COBOL_ColumnLineSum extends com.eagle.tokens.TokenSequence
				public class COBOL_ColumnLineSum : TokenSequence
				{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword SUM = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("SUM");
					public COBOL_Keyword SUM = new COBOL_Keyword("SUM");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.COBOL.Symbols.COBOL_Identifier_Reference id;
					public COBOL_Identifier_Reference id;
				}
			}
		}
	}
}
